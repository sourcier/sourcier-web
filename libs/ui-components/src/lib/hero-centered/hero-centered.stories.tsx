import { Story, Meta } from '@storybook/react';

import { HeroCentered, HeroProps } from './hero-centered';
import { heading, copy } from './hero-centered.mock';

export default {
  component: HeroCentered,
  title: 'Hero',
} as Meta;

const Template: Story<HeroProps> = (args) => <HeroCentered {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  heading,
  copy,
};
