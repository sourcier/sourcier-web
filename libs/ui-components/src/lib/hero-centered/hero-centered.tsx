import clsx from 'clsx';

export interface HeroProps {
  heading: string;
  copy?: string;
  fullHeight?: boolean;
}

export function HeroCentered({ heading, copy, fullHeight = false }: HeroProps) {
  return (
    <div
      className={clsx(
        'hero rounded-lg my-8 bg-gradient-to-t from-base-200 to-base-100 p-4 shadow-sm',
        {
          'min-h-[15vh]': !fullHeight,
          'min-h-full': fullHeight,
        }
      )}
    >
      <div className="text-center hero-content">
        <div className="max-w-md">
          <h1 className="text-5xl font-bold">{heading}</h1>
          <p className="py-6">{copy}</p>
        </div>
      </div>
    </div>
  );
}

export default HeroCentered;
