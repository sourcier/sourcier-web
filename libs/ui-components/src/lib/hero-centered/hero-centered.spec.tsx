import renderer from 'react-test-renderer';

import HeroCentered from './hero-centered';
import { heading, copy } from './hero-centered.mock';

describe('Hero', () => {
  it('should render successfully', () => {
    const tree = renderer
      .create(<HeroCentered copy={copy} heading={heading} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
