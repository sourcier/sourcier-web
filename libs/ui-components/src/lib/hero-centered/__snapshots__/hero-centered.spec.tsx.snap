// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroCentered should render successfully 1`] = `
<div
  className="hero rounded-lg my-12 bg-gradient-to-t from-base-200 to-base-100 p-4 shadow-sm min-h-[15vh]"
>
  <div
    className="text-center hero-content"
  >
    <div
      className="max-w-md lg:max-w-full"
    >
      <h1
        className="text-6xl font-bold"
      >
        Hello there
      </h1>
      Provident cupiditate voluptatem et in. Quaerat fugiat ut assumenda excepturi exercitationem quasi. In deleniti eaque aut repudiandae et a id nisi.
    </div>
  </div>
</div>
`;
